source/agent/addons/audioRanker/AudioRankerWrapper.cc

```diff
@@ -0,0 +1,151 @@
+// Copyright (C) <2019> Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "AudioRankerWrapper.h"
+#include "MediaFramePipelineWrapper.h"
+
+using namespace v8;
+
+Nan::Persistent<Function> AudioRanker::constructor;
+
+AudioRanker::AudioRanker() {};
+AudioRanker::~AudioRanker() {};
+
+NAN_MODULE_INIT(AudioRanker::Init) {
+  // Prepare constructor template
+  Local<FunctionTemplate> tpl = Nan::New<FunctionTemplate>(New);
+  tpl->SetClassName(Nan::New("AudioRanker").ToLocalChecked());
+  tpl->InstanceTemplate()->SetInternalFieldCount(1);
+
+  // Prototype
+  Nan::SetPrototypeMethod(tpl, "close", close);
+  Nan::SetPrototypeMethod(tpl, "addOutput", addOutput);
+  Nan::SetPrototypeMethod(tpl, "addInput", addInput);
+  Nan::SetPrototypeMethod(tpl, "removeInput", removeInput);
```

Comment:
why there's no removeOutput API defined that pair with removeInput?

Reply:
The number of outputs equals to the K (top). It's not designed to be reduced.